### ISSUE TITLE
RBAC: Support Active Directory

### ISSUE THREAD

**Issue by Haarolean** [kafbat/kafka-ui#54]
LdapSecurityConfig:
- verify `ActiveDirectoryLdapAuthenticationProvider` does populate authorities properly (most likely, it doesn't)
- if it doesn't, copy-paste `ActiveDirectoryLdapAuthenticationProvider` 
and (re)implement a populator (unless [#4490](https://github.com/spring-projects/spring-security/issues/4490) is resolved)

via https://github.com/provectus/kafka-ui/issues/3741

**Comment by AskarovFarkhad** [comment]
Hello everyone, please tell me how soon you plan to complete the task?
Thanks for the answer.

**Comment by AskarovFarkhad** [comment]
Hello everyone, can you tell me if there are any planned release dates for 1.1?
Thanks for the answer!

**Comment by germanosin** [comment]
> Hello everyone, can you tell me if there are any planned release dates for 1.1? Thanks for the answer!

We'll discuss dates on our next contributors meeting (Apr 10th). I hope we'll release at the beginning of May

**Comment by askfarz** [comment]
> > Hello everyone, can you tell me if there are any planned release dates for 1.1? Thanks for the answer!
> 
> We'll discuss dates on our next contributors meeting (Apr 10th). I hope we'll release at the beginning of May

Hello, is there a deadline for solving this problem? Thank you

**Comment by Haarolean** [comment]
@askfarz we'll tackle this after spring security 6.3.0 is released

**Comment by mk-raven** [comment]
There was spring security 6.3.3 released
And issue was done
https://github.com/spring-projects/spring-security/issues/4490
So, could be updates of this issue?

**Comment by wernerdv** [comment]
@Haarolean Hi, do I understand correctly that to support RBAC for AD you need to add an implementation of the LdapAuthoritiesPopulator interface for the Active Directory user?

If yes, I am ready to work on this task.

**Comment by wernerdv** [comment]
@Haarolean I've tested with this configuration and RBAC with AD works as expected:
```
auth:
  type: LDAP
spring:
  ldap:
    urls: ldap://host:port
oauth2:
  ldap:
    activeDirectory: true
    activeDirectory.domain: <domain>  

rbac:
  roles:
    - name: "<AD group>"
       clusters:
         - LOCAL
       subjects:
         - provider: ldap_ad
           type: group
           value: <AD group>
       permissions:
         - resourse: topic
           value: ".*"
           actions: [ ... ]
```

**Comment by Haarolean** [comment]
> And one more thing - you can do without `subjects.value` and `subjects.type`

that's... not intended. Let's fix this within #716.

>But it only works if rbac.roles.name is group from AD.
What makes you think so? In https://github.com/kafbat/kafka-ui/blob/main/api/src/main/java/io/kafbat/ui/service/rbac/extractor/RbacLdapAuthoritiesExtractor.java#L41 we match RBAC subjects with your AD groups and collect RBAC roles if there's a match.

**Comment by wernerdv** [comment]
https://github.com/kafbat/kafka-ui/blob/main/api/src/main/java/io/kafbat/ui/service/rbac/extractor/RbacLdapAuthoritiesExtractor.java#L39

It seems you need to handle the filtering for `Provider.LDAP_AD`.
